modules/fcrepo/iam.tf

```diff
@@ -0,0 +1,30 @@
+data "aws_iam_policy_document" "fedora_assume_role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "fedora" {
+  name               = "fedora-role"
+  path               = "/"
+  assume_role_policy = "${data.aws_iam_policy_document.fedora_assume_role.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "fedora-webtier" {
+  role       = "${aws_iam_role.fedora.name}"
+  policy_arn = "arn:aws:iam::aws:policy/AWSElasticBeanstalkWebTier"
+}
+
+resource "aws_iam_role_policy_attachment" "fedora-workertier" {
+  role       = "${aws_iam_role.fedora.name}"
+  policy_arn = "arn:aws:iam::aws:policy/AWSElasticBeanstalkWorkerTier"
+}
+
+resource "aws_iam_instance_profile" "fedora" {
+  name = "fedora-profile"
+  role = "${aws_iam_role.fedora.name}"
+}
```

Comment:
I know we did some heavy lifting of the IAM stuff in the templates but I wonder if we just want to leverage the `aws-elasticbeanstalk-ec2-role` in the environment/configuration template part in beanstalk?

Reply:
Maybe?  That could be simpler, on the one hand.  On the other, this keeps the module settings together.  I don't have a large preference either way, though I'm unfamiliar with using that atm.

Reply:
I've had a recent experience where beanstalk needed an additional policy from the iam role and therefore the home spun one needed to be updated whereas the managed one got the permission by magic. The awkward implication for using the managed role is that someone needs to manually kick off a beanstalk environment creation via their console at least once (in a brand new AWS account) to get the role created automagically for them before they can refer to it in these templates without an error. Both sides seem to have their weaknesses and strengths. I'm not sure I care either way, as long as we are consistent.